dev/gpctl/cmd/workspaces.go

```diff
@@ -28,75 +31,109 @@ var workspacesCmd = &cobra.Command{
 }
 
 func getWorkspacesClient(ctx context.Context) (*grpc.ClientConn, api.WorkspaceManagerClient, error) {
-	cfg, namespace, err := getKubeconfig()
-	if err != nil {
-		return nil, nil, err
-	}
-	clientSet, err := kubernetes.NewForConfig(cfg)
-	if err != nil {
-		return nil, nil, err
-	}
+	var addr string
+	secopt := grpc.WithInsecure()
+	if host, _ := workspacesCmd.PersistentFlags().GetString("host"); host == "" {
+		cfg, namespace, err := getKubeconfig()
+		if err != nil {
+			return nil, nil, err
+		}
+		clientSet, err := kubernetes.NewForConfig(cfg)
+		if err != nil {
+			return nil, nil, err
+		}
 
-	podLabel, err := workspacesCmd.Flags().GetString("pod")
-	if err != nil {
-		return nil, nil, err
-	}
+		podLabel, err := workspacesCmd.Flags().GetString("pod")
+		if err != nil {
+			return nil, nil, err
+		}
 
-	remotePort, err := workspacesCmd.Flags().GetString("port")
-	if err != nil {
-		return nil, nil, err
-	}
+		remotePort, err := workspacesCmd.Flags().GetString("port")
+		if err != nil {
+			return nil, nil, err
+		}
 
-	port := fmt.Sprintf("20202:%s", remotePort)
-	podName, err := util.FindAnyPodForComponent(clientSet, namespace, podLabel)
-	if err != nil {
-		return nil, nil, err
-	}
-	readychan, errchan := util.ForwardPort(ctx, cfg, namespace, podName, port)
-	select {
-	case <-readychan:
-	case err := <-errchan:
-		return nil, nil, err
-	case <-ctx.Done():
-		return nil, nil, ctx.Err()
+		port := fmt.Sprintf("20202:%s", remotePort)
+		podName, err := util.FindAnyPodForComponent(clientSet, namespace, podLabel)
+		if err != nil {
+			return nil, nil, err
+		}
+		readychan, errchan := util.ForwardPort(ctx, cfg, namespace, podName, port)
+		select {
+		case <-readychan:
+		case err := <-errchan:
+			return nil, nil, err
+		case <-ctx.Done():
+			return nil, nil, ctx.Err()
+		}
+
+		if certFromSecret, _ := workspacesCmd.Flags().GetBool("tls-from-secret"); certFromSecret {
+			certPool, err := util.CertPoolFromSecret(clientSet, namespace, "ws-manager-tls", []string{"ca.crt"})
+			if err != nil {
+				return nil, nil, xerrors.Errorf("could not load ca cert: %w", err)
+			}
+			cert, err := util.CertFromSecret(clientSet, namespace, "ws-manager-client-tls", "tls.crt", "tls.key")
+			if err != nil {
+				return nil, nil, xerrors.Errorf("could not load tls cert: %w", err)
+			}
+			creds := credentials.NewTLS(&tls.Config{
+				Certificates: []tls.Certificate{cert},
+				RootCAs:      certPool,
+				ServerName:   "ws-manager",
+			})
+
+			secopt = grpc.WithTransportCredentials(creds)
+		}
+
+		addr = "localhost:20202"
+	} else {
+		port, _ := workspacesCmd.PersistentFlags().GetString("port")
+		addr = fmt.Sprintf("%s:%s", host, port)
 	}
 
-	secopt := grpc.WithInsecure()
-	if certFromSecret, _ := workspacesCmd.Flags().GetBool("tls-from-secret"); certFromSecret {
-		certPool, err := util.CertPoolFromSecret(clientSet, namespace, "ws-manager-tls", []string{"ca.crt"})
+	if fn, _ := workspacesCmd.Flags().GetString("tls-path"); fn != "" {
```

Comment:
nit: having this as another function will reduce loc in this funtion.

Reply:
It would indeed. We've reached a point where we should pull something like this into common-go anyways